utils.js

```diff
@@ -35,126 +34,22 @@
-  }
-
-  // The callback is bound to the lifetime of the container if we have one
-  if (container != null) {
-    // Remove the unloader when the container unloads
-    container.addEventListener('unload', removeUnloader, false);
-
-    // Wrap the callback to additionally remove the unload listener
-    let origCallback = callback;
-    callback = function () {
-      container.removeEventListener('unload', removeUnloader, false);
-      origCallback();
-    };
-  }
-
-  // Wrap the callback in a function that ignores failures
-  function unloader() {
-    try {
-      callback();
-    }
-    catch(ex) {
-      // console.error(ex);
-    }
-  }
-  unloaders.push(unloader);
-
-  // Provide a way to remove the unloader
-  function removeUnloader() {
-    let index = unloaders.indexOf(unloader);
-    if (index !== -1) {
-      unloaders.splice(index, 1);
-    }
-  }
-  return removeUnloader;
-}
 
-/**
- * Apply a callback to each open and new browser windows.
- *
- * @usage watchWindows(callback): Apply a callback to each browser window.
- * @param [function] callback: 1-parameter function that gets a browser window.
- */
-function watchWindows(callback) {
-  // Wrap the callback in a function that ignores failures
-  function watcher(window) {
-    try {
-      // Now that the window has loaded, only handle browser windows
-      let {documentElement} = window.document;
-      if (documentElement.getAttribute('windowtype') === 'navigator:browser') {
-        callback(window);
-      }
-    }
-    catch(ex) {
-      // console.error(ex);
-    }
-  }
+const {Cc, Ci} = require('chrome');
 
-  // Wait for the window to finish loading before running the callback
-  function runOnLoad(window) {
-    // Listen for one load event before checking the window type
-    window.addEventListener('load', function runOnce() {
-      window.removeEventListener('load', runOnce, false);
-      watcher(window);
-    }, false);
-  }
-
-  // Add functionality to existing windows
-  let windows = Services.wm.getEnumerator(null);
-  while (windows.hasMoreElements()) {
-    // Only run the watcher immediately if the window is completely loaded
-    let window = windows.getNext();
-    if (window.document.readyState === 'complete'){
-      watcher(window);
-    // Wait for the window to load before continuing
-    } else {
-      runOnLoad(window);
-    }
-  }
-
-  // Watch for new browser windows opening then wait for it to load
-  function windowWatcher(subject, topic) {
-    if (topic === 'domwindowopened') {
-      runOnLoad(subject);
-    }
-  }
-  Services.ww.registerNotification(windowWatcher);
+/*
+ d8888b.  .d8b.  db    db db       .d88b.   .d8b.  d8888b.
+ 88  `8D d8' `8b `8b  d8' 88      .8P  Y8. d8' `8b 88  `8D
+ 88oodD' 88ooo88  `8bd8'  88      88    88 88ooo88 88   88
+ 88~~~   88~~~88    88    88      88    88 88~~~88 88   88
+ 88      88   88    88    88booo. `8b  d8' 88   88 88  .8D
+ 88      YP   YP    YP    Y88888P  `Y88P'  YP   YP Y8888D'
+*/
```

Comment:
I'm not a fan of the huge figlet headers.  Please remove them.  (If you want to add `/* PAYLOAD */` instead that would be fine by me…  🙂)